### DISCUSSION TITLE
<Your question or feedback about the GitHub Code Search and Code View products>

### DISCUSSION THREAD

**Discussion by vidisha999** [community/community#119376] [Programming Help] [answered]
### Select Topic Area

Question

### Body

**I was  doing one of the exercises of the data-quest course SQL fundamentals which is about using correlated subqueries.** 

**The question and instructions were given as**: Write a query to display all invoices where the total purchase amount (total) is greater than the average purchase amount in the same country. 

**The answer was given as** ( using a correlated subquery) : 
SELECT *
 FROM invoice AS  i1
 WHERE total > (SELECT AVG(total)
                  FROM invoice AS i2
                 WHERE i1.billing_country = i2.billing_country);*_*_

**My answer was** : 
SELECT *
FROM invoice
WHERE total > (SELECT AVG(total)
               FROM invoice
               GROUP BY billing_country)

I did not use a correlated subquery yet, the code I wrote gave the same output except for two rows which has the value of total column equal to  7.92 for two invoice_ids of France and Portugal.  I am trying to figure out what have I missed in the code that I wrote. I tried including a WHERE clause to exclude the NULL values to make sure it calculated the average total for each billing-country correctly, but it produced the same results. 

Here is a schema of the relational database for this example query
![Screen Shot 2024-04-13 at 11 08 28 PM](https://github.com/community/community/assets/166767495/4212ae3d-9d63-432d-be18-94f7d27f8a93)

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @vidisha999 !

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Code Search and Navigation category is solely related to conversations around the GitHub product Code Search and Navigation. This question would be bet received in the Programming Help category. I've gone ahead and moved it for you. Good luck!

P.S. Great question format! I love how detailed you are!

**Reply by LiteBrite82** [reply]
Is there a repository with an SQL schema you can share for localized testing?

**Reply by vidisha999** [reply]
Hi @LiteBrite82 ,
 I have not created a repository for this question, but I was able to figure out the reason why my query did not provided the accurate results and have answered it in below. 

Thanks again for the support!

**Comment by vidisha999** [comment, marked as answer]
I figured it out on later why my query was not accurate. 
The reason was, with the WHERE clause, I have used a multi-row subquery ( list of values) to compare a (one value) which is the total per unique invoice_id for a certain billing country.  

By using a multi-row subquery ( which produced the output as a list AVG (total))for each billing country), the AVG (total) of a specific billing country is not being selected when doing the comparison.
